app/grandchallenge/archives/tasks.py

```diff
@@ -35,6 +38,34 @@ def add_images_to_archive(*, upload_session_pk, archive_pk, interface_pk=None):
         item.values.set([civ])
 
 
+@shared_task
+def add_images_to_archive_item(
+    *, upload_session_pk, archive_item_pk, interface_pk=None
+):
+    archive_item = ArchiveItem.objects.get(pk=archive_item_pk)
+    interface = ComponentInterface.objects.get(pk=interface_pk)
+
+    civ_pks_to_remove = set()
+    civ_pks_to_add = set()
+
+    civ = archive_item.values.filter(interface=interface).first()
+    if civ:
+        civ_pks_to_remove.add(civ.pk)
+
+    civ = ComponentInterfaceValue.objects.create(interface=interface)
```

Comment:
Maybe call this `new_civ` to differentiate it from the one(s) being removed above?